### PR TITLE
Team assignment for Artefact signoff/review

### PR DESCRIPTION
**Pull request by omar-selo**
This resolves [RTW-208](https://warthogs.atlassian.net/browse/RTW-208)

The way this works is that there is an endpoint to create users that takes as input launchpad email, then queries launchpad to get handle and name, then stores that all in the database. This new endpoint to create users is not shown in docs as it's supposed to be internal.

When a new test execution is received for an artefact that has no one assigned to it, a random user will be assigned to this artefact. (random instead of round robin cause round robin is not simple given multiple TO units).

There is also a new endpoint to change the assigned user id. This endpoint is also not shown in docs as it's internal too.

[RTW-208]: https://warthogs.atlassian.net/browse/RTW-208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by andrejvelichkovski — DISMISSED** [review]
Overall it looks good to me and I think it is ready to be merged to main.

I just noticed two small details: the first one for the duplicated class with the same name, and the second one for sending more meaningful error message in case the LP user couldn't be found by email.

File: backend/test_observer/controllers/artefacts/models.py [hunk 1]
```diff
@@ -26,6 +26,13 @@
 )
 
 
+class UserDTO(BaseModel):
```
[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/test_observer/controllers/artefacts/models.py]: Ah good catch. I deleted the minimized DTO and used one object for both. We should always return the same representation of the same object

File: backend/test_observer/controllers/users/users.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+# include_in_schema=False to hide this endpoints from docs as its internal use
+@router.post("/", response_model=UserDTO, include_in_schema=False)
+def add_user(
+    request: CreateUserRequest,
+    db: Session = Depends(get_db),
+    launchpad_api: LaunchpadAPI = Depends(LaunchpadAPI),
+):
+    launchpad_user = launchpad_api.get_user_by_email(request.launchpad_email)
```
[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/test_observer/controllers/users/users.py:20]: Good idea

**Review by andrejvelichkovski — DISMISSED** [review]
Thanks for addressing the suggestions quickly!

**Pull request merged**